core/trino-main/src/main/java/io/trino/operator/aggregation/AggregationFromAnnotationsParser.java

```diff
@@ -72,101 +70,112 @@ public static ParametricAggregation parseFunctionDefinitionWithTypesConstraint(C
+            Class<?> stateClass,
+            List<AggregationImplementation> exactImplementations,
+            List<AggregationImplementation> nonExactImplementations)
     {
-        ParametricImplementationsGroup.Builder<AggregationImplementation> implementationsBuilder = ParametricImplementationsGroup.builder();
-        AggregationHeader header = parseHeader(aggregationDefinition);
-        boolean deprecated = aggregationDefinition.getAnnotationsByType(Deprecated.class).length > 0;
+        ImmutableList.Builder<ParametricAggregation> functions = ImmutableList.builder();
+
+        // create a separate function for each exact implementation
+        for (AggregationImplementation exactImplementation : exactImplementations) {
+            functions.add(new ParametricAggregation(
+                    signatureWithName(name, exactImplementation.getSignature()),
+                    header,
+                    stateClass,
+                    ParametricImplementationsGroup.of(exactImplementation).withAlias(name)));
+        }
 
-        for (Class<?> stateClass : getStateClasses(aggregationDefinition)) {
-            Method combineFunction = getCombineFunction(aggregationDefinition, stateClass);
-            Method outputFunction = getOnlyElement(getOutputFunctions(aggregationDefinition, stateClass));
-            for (Method inputFunction : getInputFunctions(aggregationDefinition, stateClass)) {
-                Optional<Method> removeInputFunction = getRemoveInputFunction(aggregationDefinition, inputFunction);
-                AggregationImplementation implementation = parseImplementation(aggregationDefinition, header, stateClass, inputFunction, removeInputFunction, outputFunction, combineFunction);
-                implementationsBuilder.addImplementation(implementation);
-            }
+        // if there are non-exact functions, create a single generic/calculated function using these implementations
+        if (!nonExactImplementations.isEmpty()) {
+            ParametricImplementationsGroup.Builder<AggregationImplementation> implementationsBuilder = ParametricImplementationsGroup.builder();
+            nonExactImplementations.forEach(implementationsBuilder::addImplementation);
+            ParametricImplementationsGroup<AggregationImplementation> implementations = implementationsBuilder.build();
+            functions.add(new ParametricAggregation(
+                    signatureWithName(name, implementations.getSignature()),
+                    header,
+                    stateClass,
+                    implementations.withAlias(name)));
         }
 
-        ParametricImplementationsGroup<AggregationImplementation> implementations = implementationsBuilder.build();
-        return new ParametricAggregation(signatureWithName(header.getName(), implementations.getSignature()), header, implementations, deprecated);
+        return functions.build();
+    }
+
+    private static boolean isExact(Signature signature)
```

Comment:
This needs a better name. If you can you describe what concept this represents I can help come up with another name.